machine/v1/types_alibabaprovider.go

```diff
@@ -0,0 +1,340 @@
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// InstanceID is the instance ID of the machine created in alibabacloud
+	// +optional
+	InstanceID *string `json:"instanceId,omitempty"`
+
+	// InstanceState is the state of the alibabacloud instance for this machine
+	// +optional
+	InstanceState *string `json:"instanceState,omitempty"`
+
+	// Conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// SystemDiskProperties contains the information regarding the system disk including performance, size, name, and category
+type SystemDiskProperties struct {
+	// Category is the category of the system disk.
+	// Valid values:
+	// cloud_essd: ESSD. When the parameter is set to this value, you can use the SystemDisk.PerformanceLevel parameter to specify the performance level of the disk.
+	// cloud_efficiency: ultra disk.
+	// cloud_ssd: standard SSD.
+	// cloud: basic disk.
+	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
+	// Currently for non-I/O optimized instances of retired instance types, the default is `cloud`.
+	// Currently for other instances, the default is `cloud_efficiency`.
+	// +kubebuilder:validation:Enum="cloud_efficiency"; "cloud_ssd"; "cloud_essd"; "cloud"
+	// +optional
+	Category string `json:"category,omitempty"`
+
+	// PerformanceLevel is the performance level of the ESSD used as the system disk.
+	// Valid values:
+	//
+	// PL0: A single ESSD can deliver up to 10,000 random read/write IOPS.
+	// PL1: A single ESSD can deliver up to 50,000 random read/write IOPS.
+	// PL2: A single ESSD can deliver up to 100,000 random read/write IOPS.
+	// PL3: A single ESSD can deliver up to 1,000,000 random read/write IOPS.
+	// Empty value means no opinion and the platform chooses a default, which is subject to change over time.
+	// Currently the default is `PL1`.
+	// For more information about ESSD performance levels, see ESSDs.
+	// +kubebuilder:validation:Enum="PL0"; "PL1"; "PL2"; "PL3"
+	// +optional
+	PerformanceLevel string `json:"performanceLevel,omitempty"`
+
+	// Name is the name of the system disk. If the name is specified the name must be 2 to 128 characters in length. It must start with a letter and cannot start with http:// or https://. It can contain letters, digits, colons (:), underscores (_), and hyphens (-).
+	// Empty value means the platform chooses a default, which is subject to change over time.
+	// Currently the default is `""`.
+	// +kubebuilder:validation:MaxLength=128
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// Size is the size of the system disk. Unit: GiB. Valid values: 20 to 500.
+	// The value must be at least 20 and greater than or equal to the size of the image.
+	// The default value is 40 or the size of the image, depending on whichever is greater.
```

Comment:
subject to change in the future?

Reply:
I believe the installer is setting this value to to 120. As noted this fails if the disk size is not larger than the image size. I will add the verbiage that the API team desires.

Reply:
@deads2k Would you like me to update the documentation here to include the verbiage?
```
	// Empty value means no opinion and the platform chooses the a default, which is subject to change over time.
	// Currently the default is `40`
```
If so I will update.